python/pyproject.toml

```diff
@@ -17,7 +17,8 @@
 
 [build-system]
 requires = [
+    "boto3==1.19.2",
```

Comment:
what's the reason for the boto pin here? we may want to consider not having hard dependencies on boto and instead guard the imports in the S3 implementations when boto is imported

Reply:
Thanks @TGooch44, I removed the pin and I'll update the PR with the import guards. I also moved the boto3 requirement to an `s3` extras block in `setup.py`. I'll add an item to the next python sync to discuss this but I think there's value in having optional dependency sets that let users only install what they need. For example a user doesn't get boto3 unless they do `pip3 install py-iceberg[s3]`. This will let those going with their own custom file io and custom catalog to just install the core components with `pip3 install py-iceberg`.